### ISSUE TITLE
[C++][Parquet] Int96 reader in parquet_arrow uses size of Int96Type instead of Int96

### ISSUE THREAD

**Issue by asfimport** [apache/arrow#42692]
This produces a segfault when reading `alltypes_plain.parquet` with parquet::arrow. I will see if I can reproduce with a test case.

**Reporter**: [Wes McKinney](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=wesm) / @wesm
**Assignee**: Wes McKinney](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=wesm) / @wesm

<sub>**Note**: *This issue was originally created as [PARQUET-995](https://issues.apache.org/jira/browse/PARQUET-995). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

**Comment by asfimport** [comment]
[Wes McKinney](https://issues.apache.org/jira/browse/PARQUET-995?#comment-16015758) / @wesm:
PR: https://github.com/apache/parquet-cpp/pull/337

**Comment by asfimport** [comment]
[Uwe Korn](https://issues.apache.org/jira/browse/PARQUET-995?#comment-16016349) / @xhochy:
Issue resolved by pull request 337
<https://github.com/apache/parquet-cpp/pull/337>

**Comment by asfimport** [comment]
[YE](https://issues.apache.org/jira/browse/PARQUET-995?#comment-16019758) / @advancedxy:
Hi, @wesm how is this issue found? 

I was haunted by this bug all day today. In our application, the segfault showed up in the core dump pointed to some innocent code(not related to parquet) due to the bad memory access. I tried everything and until I found this issue,  it occurred to me this might be the root cause. And indeed the problem is solved when I merged the latest parquet code.
